spec/unit/middlewares/locale_spec.rb

```diff
@@ -9,29 +9,35 @@
   let(:site)            { instance_double('Site', default_locale: :de, locales: [:de, :fr, :en]) }
   let(:url)             { 'http://models.example.com' }
   let(:app)             { ->(env) { [200, env, 'app'] } }
-  let(:services)        { instance_double('Services', :locale= => 'en') }
+  let(:cookie_lang)     { nil }
+  let(:cookie_service)  { instance_double('Cookie Service', :get => cookie_lang) }
+  let(:services)        { instance_double('Services', :locale= => 'en', :cookie => cookie_service) }
   let(:middleware)      { Locomotive::Steam::Middlewares::Locale.new(app) }
-  let(:session)         { {} }
   let(:accept_language) { '' }
+  let(:expected_lang)   { :de }
```

Comment:
by convention, we don't use the `let` keyword to store expected results. It's better to read the expected result in the test itself.